subiquity/ui/views/filesystem/filesystem.py

```diff
@@ -168,7 +170,9 @@ def refresh_model_inputs(self):
                         ]
             actions = [(_("Unmount"), mi.mount.can_delete(), "unmount")]
             menu = ActionMenu(actions)
-            connect_signal(menu, "action", self._mount_action, mi.mount)
+            connect_async_signal(
+                menu, "action", self._mount_action, user_args=[mi.mount]
```

Comment:
This sort of thing is interesting. Are the semantics of async callbacks what we want here? In that the task of updating the model happens in the background to some extent and then the UI is redrawn some (in theory) arbitrary time later. Is that what we want? What _do_ we want? Do we want to do TuiApplication._wait_with_indication games to block the UI for MAX_BLOCK_TIME and then do something like grey out the widget if the API call takes longer than that?

Am I overthinking this?

Reply:
Yes, I think we should not update the UI until the API call returns. Whether we should _wait_with_indication is a good point to consider implementing at some point (so far, I don't have any specific use case in mind where the API would be that slow to finish).

I think redrawing the UI after the background task finishes would be ideal. We'd need access to view (or the urwid handler) to request a redraw though ; so I need to think about what that means.